src/movies.js

```diff
@@ -1,16 +1,134 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies) {
+    let answer = movies.map(function(movie){
+        return movie.director;
+    })
+    return answer;
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+function getUniqueList(movies) {
+    let list = [];
+    for (let i = 0; i <movies.length; i++) {
+        let director = movies[i].director;
+        if (list.indexOf(director) < 0) {
+            list.push(director);
+        }
+    }
+    return list;
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies) {
+    let drama;
+      let spiel = movies.filter(function(movie) {
+          for (let i = 0; i < movie.genre.length; i++) {
```

Comment:
Using a loop works fine here, but you could avoid it if you look into the .includes method for the genre key. The method can check if a string exist inside an array.